common/templates/general.go

```diff
@@ -33,19 +35,38 @@ func Dictionary(values ...interface{}) (map[interface{}]interface{}, error) {
 }
 
 func StringKeyDictionary(values ...interface{}) (SDict, error) {
+	
 	if len(values) == 1 {
-		switch t := values[0].(type) {
-		case SDict:
-			return t, nil
-		case map[string]interface{}:
-			mapCopy := make(map[string]interface{})
-			for key, value := range t {
-				mapCopy[key] = value
-			}
-			return SDict(mapCopy), nil		
-		default:
-			return nil, errors.New("invalid dict call")		
-		}	
+		val, isNil := indirect(reflect.ValueOf(values[0]))
+		if isNil || values[0] == nil {
+			return nil, errors.New("Sdict: nil value passed")
+		} 		
+
+		if val.Type().String() == "templates.SDict" {	
```

Comment:
change to 
```go
if sdict, ok :=val.(SDict); ok {
    return sdict, nil
```

Reply:
I think you meant val.Interface().(SDict) because val is reflect.Value and not interface and cannot be typecasted. I have updated according to my understanding.